src/AbstractEnumeration.php

```diff
@@ -24,7 +24,9 @@
         $reflector = new ReflectionClass(get_called_class());
 
         foreach ($reflector->getConstants() as $key => $value) {
```

Comment:
I haven't tested this, but I think we can make this better by using `getReflectionConstants` directly:

```suggestion
         foreach ($reflector->getReflectionConstants() as $constant) {
```